philips_hue.py

```diff
@@ -0,0 +1,135 @@
+
+    Attributes:
+        bridge_ip_address: IP address of the Hue bridge system to connect to.
+        username: Authorized user string to make API calls.
+    """
+    def __init__(self, bridge_ip_address, username):
+        self._bridge_ip_address = bridge_ip_address
+        self._username = username
+
+
+    @property
+    def bridge_ip_address(self):
+        return self._bridge_ip_address
+
+
+    @property
+    def username(self):
+        return self._username
+
+
+    def set_color(self, light_id, hue):
+        """Sets the color of the light to a specified hue value.
+
+        Args:
+            light_id:  The id to pass to the Philips Hue API to
+                specify the light to set a color for.
+            hue: Hue of the light. Takes values from 0 to 65535.
+                Programming 0 and 65535 would mean that the light will resemble
+                the color red, 25500 for green and 46920 for blue.
+
+        Returns:
+            HTTP Response from the Philips Hue API.
+        """
+        response = requests.put(url=f'http://{self._bridge_ip_address}/api/{self._username}/lights/{light_id}/state',
+                                data=json.dumps({"on": True, "hue": hue}))
+        if response.status_code != 200:
+            raise BadAPIRequestError(response.text)
+        return response
+
+
+# TODO(https://github.com/googleinterns/cloud-monitoring-notification-delivery-integration-sample-code/issues/10):
+# Currently specific to Philips Hue, but will be generalized to trigger
+# whatever notification system the client chooses.
+def trigger_light_from_monitoring_notification(client, notification, light_id):
+    """Changes the color of a Philips Hue light based on a monitoring notification
+    and returns the response.
+
+    The color of the Philips Hue light is set through the client,
+    which makes an HTTP PUT request to the Philips Hue API.
+    Sets the color of the light to red if the incident is open and green if the incident is closed.
+
+    Args:
+        client: The PhilipsHueClient object to trigger a response from.
+        notification: A dictionary containing the notification data.
+        light_id: The id of the light to set the color for.
+
+    Returns:
+        The state of the provided Philips Hue after the monitoring notification
+        was translated into a light signal, and then forwarded to the bulb.
+        One of PhilipsHueState.
+
+    Raises:
+        UnknownConditionStateError: If the incident state is not open or closed.
+        BadAPIRequestError: If there was an error in calling the Philips Hue API.
+    """
+    try:
+        condition_state = notification["incident"]["condition"]["state"]
+    except KeyError:
+        raise NotificationParseError("Notification is missing required dict key")
+
+    if condition_state == "open":
+        open_state = PhilipsHueState.OPEN
+        client.set_color(light_id, open_state.value)  # set to red
+        return repr(open_state)
```

Comment:
Why not just return `open_state`. Then, the caller can call repr() if they want a representation string. Or open_state.name, or open_state.value.